src/tools/tidy/src/fluent_used.rs

```diff
@@ -0,0 +1,36 @@
+//! Checks that all Fluent messages appear at least twice
+
+use crate::walk::{filter_dirs, walk};
+use regex::Regex;
+use std::collections::HashMap;
+use std::path::Path;
+
+lazy_static::lazy_static! {
+    static ref WORD: Regex = Regex::new(r"\w+").unwrap();
+}
+
+fn filter_used_messages(
+    contents: &str,
+    msgs: &mut HashMap<String, String>,
+    unused_msgs: &mut HashMap<String, String>,
+) {
+    let mut matches = WORD.find_iter(contents);
+    while let Some(name) = matches.next() {
+        if let Some((name, filename)) = msgs.remove_entry(name.as_str()) {
+            unused_msgs.insert(name, filename);
+        } else {
+            unused_msgs.remove(name.as_str());
+        }
+    }
+}
+
+pub fn check(path: &Path, msgs: &mut HashMap<String, String>, bad: &mut bool) {
```

Comment:
```suggestion
pub fn check(path: &Path, all_defined_msgs: &mut HashMap<String, String>, bad: &mut bool) {
```
This makes it clearer that we mean *all* the message there were defined here, some of which may be unused.